src/Storefront/Resources/app/storefront/src/plugin-system/plugin.manager.js

```diff
@@ -266,18 +266,92 @@ class PluginManagerSingleton {
         }
 
         if (typeof selector === 'string') {
-            selector = document.querySelectorAll(selector);
+            selector = PluginManagerSingleton._queryElements(selector, pluginClass, options, pluginName);
+            if (!selector) {
```

Comment:
If the selector is always iterable you can remove the additional return statement.